content/browser/ephemeral_storage_partition.h

```diff
@@ -0,0 +1,68 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CONTENT_BROWSER_EPHEMERAL_STORAGE_PARTITION_H_
+#define BRAVE_CONTENT_BROWSER_EPHEMERAL_STORAGE_PARTITION_H_
+
+#include <string>
+#include <utility>
+
+#include "content/public/browser/storage_partition.h"
+#include "services/network/public/mojom/network_context.mojom.h"
+#include "base/memory/ref_counted.h"
+
+namespace content {
+
+using EphemeralStorageParitionMapKey =
+    std::pair<content::BrowserContext*, std::string>;
+
+class StoragePartitionImpl;
+
+// EphemeralStoragePartition manages an in-memory NetworkingContext to provide
+// ephemeral cookie storage.
+class EphemeralStoragePartition
```

Comment:
If this is moved to `content/public` then we can avoid patching BrowserContext, right?